packages/grid/x-grid/src/tests/editRows.XGrid.test.tsx

```diff
@@ -139,6 +139,17 @@ describe('<XGrid /> - Edit Rows', () => {
     expect(cell.textContent).to.equal('');
   });
 
+  it('should not allow to delete a cell directly if it is not editable', () => {
+    render(<TestCase isCellEditable={()=> false} />);
+    const cell = getCell(1, 0);
+    cell.focus();
+
+    expect(cell.textContent).to.equal('Adidas');
+    fireEvent.keyDown(cell, { key: 'Delete' });
+    expect(cell).to.not.have.class('MuiDataGrid-cellEditable');
```

Comment:
See https://github.com/mui-org/material-ui/pull/24884 for the best English semantic:

```suggestion
    expect(cell).not.to.have.class('MuiDataGrid-cellEditable');
```